### DISCUSSION TITLE
Warmstarting LP

### DISCUSSION THREAD

**Discussion by Iain-R** [coin-or/Clp#276] [Q&A] [answered]
Hello all, 
I was looking for assistance or guidance on warmstarting an LP with a feasible solution. 
I can do this in Gurobi by setting primal variable values, as follows :
``` cpp 
void LPWrapper::setWarmStartValues(std::vector<std::pair<int, double>>& columnStarts) {
        for (auto& startVal : columnStarts) {
            int variable = m_variables.at(startVal.first);
            double value = startVal.second;
            m_model.getVar(variable).set(GRB_DoubleAttr_PStart, value);
      }
}
```

Can I do something similar in COIN CLP  ? 
Thank you.
Iain

**Comment by jhmgoossens** [comment, marked as answer]
Via [OsiClpSolverInterface there is WarmStart](https://github.com/coin-or/Clp/blob/master/src/OsiClp/OsiClpSolverInterface.cpp#L1866C59) functionality available. This, however, [represents a basis](https://github.com/coin-or/Clp/blob/master/src/OsiClp/OsiClpSolverInterface.cpp#L1888) not just a solution.
Hope this points you in the right direction.

**Reply by Iain-R** [reply]
Thank you for your quick reply,
I am solving a series of LP problems where the solution at iteration K is a feasible solution of iteration K+1. The constraint sets may be different between iterations. With this in mind, is there a way to convert a feasible solution to a feasible starting basis?

**Reply by jhmgoossens** [reply]
Unfortunately, I'm no expert in this area, but this is what [ClpModel's setColSolution](https://github.com/coin-or/Clp/blob/master/src/ClpModel.hpp#L529) or [OsiClpSolverInterface's setColSolution](https://github.com/coin-or/Clp/blob/master/src/OsiClp/OsiClpSolverInterface.cpp#L4242) are for. For an example, see [CbcSolver3.cpp](https://github.com/coin-or/Cbc/blob/master/examples/CbcSolver3.cpp#L126).
